ibis/expr/types/relations.py

```diff
@@ -1128,8 +1166,29 @@ def aggregate(
                 else:
                     metrics[metric.name] = metric
 
-        # construct the aggregate node
-        agg = ops.Aggregate(node, groups, metrics).to_expr()
+        keys = frozendict(
```

Comment:
This bit is far too optimistic and very proof-of-concept. More unit tests are required to throw some chaos at this code.